packages/pipeline/src/parsers/ddex_orders/index.ts

```diff
@@ -0,0 +1,73 @@
+import { BigNumber } from '@0x/utils';
+import * as R from 'ramda';
+
+import { DdexMarket, DdexOrder, DdexOrderbook } from '../../data_sources/ddex';
+import { TokenOrderbookSnapshot as TokenOrder } from '../../entities';
+import { OrderType } from '../../types';
+
+/**
+ * Marque function of this file.
+ * 1) Takes in orders from an orderbook,
+ * 2) Aggregates the orders based on price point,
+ * 3) For each aggregated order, forms a TokenOrder entity with market data and
+ * other information attached.
+ * @param ddexOrderbook A raw orderbook that we pull from the Ddex API.
+ * @param ddexMarket An object containing market data also directly from the API.
+ * @param retrievalTimestamp Time at which the orders for the market were pulled.
+ * @param source The exchange where these orders are placed. In this case 'ddex'.
+ */
+export function parseDdexOrders(ddexOrderbook: DdexOrderbook, ddexMarket: DdexMarket, retrievalTimestamp: BigNumber, source: string): TokenOrder[] {
+    const aggregatedBids = aggregateOrders(ddexOrderbook.bids);
+    const aggregatedAsks = aggregateOrders(ddexOrderbook.asks);
+    // Might want to shift 'bid', 'ask' enums to 0x/types and retrieve from there.
+    const parsedBids = R.map(parseDdexOrder(ddexMarket, retrievalTimestamp, 'bid', source), aggregatedBids) ;
+    const parsedAsks = R.map(parseDdexOrder(ddexMarket, retrievalTimestamp, 'ask', source), aggregatedAsks);
+    return parsedBids.concat(parsedAsks);
+}
+
+/**
+ * Aggregates orders by price point for consistency with other exchanges.
+ * Querying the Ddex API at level 3 setting returns a breakdown of
+ * individual orders at each price point. Other exchanges only give total amount
+ * at each price point. Returns an array of <price, amount> tuples.
+ * @param ddexOrders A list of Ddex orders awaiting aggregation.
+ */
+export function aggregateOrders(ddexOrders: DdexOrder[]): Array<[string, number]> {
+    const aggregatedPricePoints =  ddexOrders.reduce((orders: Map<string, number>, order: DdexOrder): Map<string, number> => {
+        const price = order.price;
+        const amount = orders.get(price) || 0;
+        orders.set(price, amount + Number(order.amount));
+        return orders;
+    }, new Map());
+    return [ ...aggregatedPricePoints.entries()];
+}
+
+const parseDdexOrder = R.curry((ddexMarket: DdexMarket, retrievalTimestamp: BigNumber, orderType: string, source: string, ddexOrder: [string, number]): TokenOrder => {
```

Comment:
It looks like all this function does is call `_parseDdexOrder`. Can we just remove it and call `_parseDdexOrder` directly?